### PR TITLE
Replace AssemblyVersion with Version

### LINKED ISSUE #2393: [Bug] Pin IdentityModel version for own dependencies
This will help prevent the mismatch of M.IM versions...all M.IM versions need to be the same.

### PR DESCRIPTION
**Pull request by FuPingFranco**
## Description

Replace dependency on updateAssemblyInfo.ps1 with Version property and ADO variables. In order to bring our build and release experience across our repos to a unified process, we need to adopt the way versioning is done our other repos.

Test runs:

WisonVersion = ' ' (https://identitydivision.visualstudio.com/IDDP/_build/results?buildId=1257450&view=artifacts&pathAsName=false&type=publishedArtifacts )

WisonVersion = 7.3.3-preview-test-123 (https://identitydivision.visualstudio.com/IDDP/_build/results?buildId=1257449&view=artifacts&pathAsName=false&type=publishedArtifacts)

WisonVersion = 7.3.3 (https://identitydivision.visualstudio.com/IDDP/_build/results?buildId=1257451&view=artifacts&pathAsName=false&type=publishedArtifacts)

Fixes https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/2393

### CODE REVIEW

File: buildConfiguration.xml [hunk 1]
```diff
@@ -2,8 +2,6 @@
   <dotnetArchitecture>x64</dotnetArchitecture>
   <nugetVersion>3.5.0-rc-1285</nugetVersion>
```
Comment (jmprieur) [inline buildConfiguration.xml:3]: Is `nugetVersion` used?

Comment (jmprieur) [reply buildConfiguration.xml:3]: More generally is the buildConfiguration.xml used at all anywhere?

File: .github/workflows/dotnetcore.yml [hunk 2]
```diff
@@ -30,9 +30,6 @@ jobs:
       with:
         dotnet-version: 8.0.x
 
-    - name: update versions
-      run: .\updateAssemblyInfo.ps1
-
     # Build and test
```
Comment (jmprieur) [inline .github/workflows/dotnetcore.yml]: the dotnet restore is not needed. It's done by dotnet test

Comment (jmprieur) [reply .github/workflows/dotnetcore.yml]: also, we should repalce the run .\pack.ps1 by dotnet pack --no-restore --no-build

**Review by jmprieur — APPROVED** [review]
Thanks @FuPingFranco 
This looks good.

I think you can still update the github flow, and remove the pack.ps1 script and the buildconfiguration.xml file.

Did you update the process?

File: buildConfiguration.xml (hunk 1 above)
Comment (FuPingFranco) [reply buildConfiguration.xml:3]: @jmprieur nugetVersion is used mainly in pack.ps1 and build.ps1 so we could remove it as well potentially. As for buildConfiguration.xml we do use it when running the VerifySigning.ps1 among a couple of other scripts so, I created a separate task to track this https://identitydivision.visualstudio.com/Engineering/_boards/board/t/Auth%20Client%20-%20AuthNZ%20-%20Customer%20trust/Backlog%20items/?workitem=2850069 and look into it in detail so, I can focus this PR on versioning. I can pick this up on my next OCE rotation along with SAL scripts as well. Thanks for the feedback JM!

File: .github/workflows/dotnetcore.yml (hunk 2 above)
Comment (FuPingFranco) [reply .github/workflows/dotnetcore.yml]: @jmprieur , Good catch, I've removed them. Thanks JM!

**Comment by FuPingFranco** [comment]
> Thanks @FuPingFranco This looks good.
> 
> I think you can still update the github flow, and remove the pack.ps1 script and the buildconfiguration.xml file.
> 
> Did you update the process?

I updated the github flow and created a separate task to look into all scripts post this changes. Also, working on documentation on the process and will share it asap. Thank you for all the help @jmprieur :)

**Review by jmprieur — APPROVED** [review]
Awesome.
Thanks @FuPingFranco 
Great job!

**Pull request merged**